plugins/spectate/spectate.cpp

```diff
@@ -78,14 +77,41 @@ static PersistentDataItem pconfig;
 
 DFhackCExport command_result plugin_enable(color_ostream &out, bool enable);
 command_result spectate (color_ostream &out, std::vector <std::string> & parameters);
-#define COORDARGS(id) id.x, (id).y, id.z
+#define COORDARGS(id) id.x, id.y, id.z
+
+namespace DFHack {
+    namespace Units {
+        bool isInBox(df::unit* u,
+                     int16_t x1, int16_t y1, int16_t z1,
+                     int16_t x2, int16_t y2, int16_t z2) {
+            if (u->pos.x >= x1 && u->pos.x <= x2) {
+                if (u->pos.y >= y1 && u->pos.y <= y2) {
+                    if (u->pos.z >= z1 && u->pos.z <= z2) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+    }
+}
 
 namespace SP {
     bool following_dwarf = false;
     df::unit* our_dorf = nullptr;
     int32_t timestamp = -1;
     std::default_random_engine RNG;
 
+    void DebugUnitVector(std::vector<df::unit*> units) {
+        for (auto unit : units) {
+            INFO(plugin).print("[id: %d]\n animal: %d\n hostile: %d\n visiting: %d\n",
```

Comment:
you can change the verbosity level to `DEBUG` and wrap the whole for loop in an
```
if (DBG_NAME(plugin).isEnabled(DebugCategory::LDEBUG)) {
...
}
```
then you can just leave the debug statements in instead of commenting out the call to `DebugUnitVector`

Reply:
Either way a code change and rebuild is required to run that code. I didn't provide debug commands for this cause it's super simple (compared to channel-safely).

Reply:
why would a code change be required? and you shouldn't need to provide debug commands at all -- `debugfilter` provides that interface.

Reply:
link, or elaborate? `debugfilter`?